Cython/Compiler/Nodes.py

```diff
@@ -3409,7 +3415,7 @@ def generate_function_definitions(self, env, code):
         code.put_label(code.return_label)
         for entry in lenv.var_entries:
             if entry.is_arg and entry.type.is_pyobject:
-                code.put_var_decref(entry)
+                code.put_var_xdecref(entry)
```

Comment:
I think this is because `put_var_decref` historically always put `xdecref`. Again, I'll see if a bit more intelligence can be applied here, but this isn't a degradation.